common/test/acceptance/pages/studio/settings_certificates.py

```diff
@@ -312,59 +351,108 @@ def click_create_certificate_button(self):
         """
         Add signatory to certificate
         """
-        self.find_css('.action-add-signatory').first.click()
+        self.q(css='.action-add-signatory').first.click()
 
     def click_edit_certificate_button(self):
         """
         Open editing view for the certificate.
         """
-        self.find_css('.action-edit .edit').first.click()
+        self.q(css='.action-edit .edit').first.click()
 
     def click_cancel_edit_certificate(self):
         """
         Cancel certificate editing.
         """
-        self.find_css('.action-secondary').first.click()
+        self.q(css='.action-secondary').first.click()
 
     def click_certificate_details_toggle(self):
         """
         Expand/collapse certificate configuration.
         """
-        self.find_css('a.detail-toggle').first.click()
+        self.q(css='a.detail-toggle').first.click()
 
     def click_delete_certificate_button(self):
         """
         Remove the first (possibly the only) certificate from the set
         """
         self.wait_for_certificate_delete_button()
-        self.find_css('.actions .delete.action-icon').first.click()
+        self.q(css='.actions .delete.action-icon').first.click()
 
 
-class Signatory(object):
+class SignatorySectionPage(CertificatesPage):
     """
-    Signatory page object wrapper
+    SignatorySectionPage is the signatory section within CertificatesSection, There might be multiple signatories
+    in a certificate section so this section object can be used to used to identify unique section and apply
+    operations on it.
     """
     def __init__(self, certificate, prefix, mode, index):
+        """
+        Initialize SignatorySection Page
+
+        :param certificate: Container Section Page Object of the Signatory section
+        :param prefix: css selector of the container element
+        :param index: index of section in the signatory list on the page
+        :param mode: 'details' or 'edit', showing whether signatory is being displayed or edited
+
+        :return:
+        """
         self.certificate = certificate
         self.prefix = prefix
         self.index = index
         self.mode = mode
 
+        super(SignatorySectionPage, self).__init__(self.certificate.browser, **self.certificate.course_info)
+
+    def is_browser_on_page(self):
+        """
+        Verify that the browser is on the page and it is not still loading.
+        """
+        EmptyPromise(
+            lambda: self.certificate.q(css=".signatory-details-list, .signatory-edit-list").present,
+            'Wait for Signatory Section to load.'
+        ).fulfill()
+        return True
+
+    @property
+    def url(self):
+        """
+        Construct a URL to the page section within the certificate section page.
+        """
+        # The url is same as CertificateSectionPage url for now.
+        return self.certificate.url
+
+    @unguarded
+    def get_selector(self, css=''):
```

Comment:
I think this is another method that becomes unnecessary now that you're using a proper PageObject.